src/slices/userFilterSlice.ts

```diff
@@ -1,29 +1,65 @@
 import { createSlice, PayloadAction } from "@reduxjs/toolkit";
-import { UserType } from "../types";
+// import { UserType } from "../types";
+import { userApi } from "../quries";
+import { map } from "lodash"
+import { dayjs } from "../vendor";
 
+const dateFormat = "YYYY-M-D";
 interface UserState {
- users: UserType[];
- filteredUsers: UserType[];
- selectedGenders: string[];
+  // users: UserType[];
+  // filteredUsers: UserType[];
+  selectedGenders: string[];
+  selectedAgeRange: number[];
+  usernames: string[];
+  selectedDateOfBirthFilter: string[];
+  nameFilters: object[];
+  // loading: boolean;
+  // error: string | null;
 }
 
 const initialState: UserState = {
- users: [],
- filteredUsers: [],
- selectedGenders: ['Male', 'Female'],
```

Comment:
keep this constant data inside `constants/GENDER.ts` and import to filters and slices, then we don't need duplicate this code and easy to maintain